jmix-search/search-starter/src/main/java/io/jmix/autoconfigure/search/EnqueueingSessionProcessingScheduleAutoConfiguration.java

```diff
@@ -19,46 +19,71 @@
 import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.ApplicationContext;
 import org.springframework.context.annotation.Bean;
 import org.springframework.boot.autoconfigure.AutoConfiguration;
 import org.springframework.context.annotation.Import;
 
 @AutoConfiguration
 @Import(SearchConfiguration.class)
 @ConditionalOnClass(Job.class)
-@ConditionalOnProperty(name = "jmix.search.use-default-enqueueing-session-processing-quartz-configuration", matchIfMissing = true)
 public class EnqueueingSessionProcessingScheduleAutoConfiguration {
 
     private static final Logger log = LoggerFactory.getLogger(EnqueueingSessionProcessingScheduleAutoConfiguration.class);
 
+    public static final String JOB_NAME = "EnqueueingSessionProcessing";
+
+    public static final String JOB_TRIGGER_NAME = "EnqueueingSessionProcessingTrigger";
+
+    public static final String JOB_GROUP = "DEFAULT";
+
     @Autowired
     protected SearchProperties searchProperties;
 
+    @Autowired
+    protected ApplicationContext applicationContext;
+
     @Bean("search_EnqueueingSessionProcessingJob")
+    @ConditionalOnProperty(name = "jmix.search.use-default-enqueueing-session-processing-quartz-configuration", matchIfMissing = true)
     JobDetail enqueueingSessionProcessingJob() {
         return JobBuilder.newJob()
                 .ofType(EnqueueingSessionProcessingJob.class)
                 .storeDurably()
-                .withIdentity("EnqueueingSessionProcessing")
+                .withIdentity(JOB_NAME)
                 .build();
     }
 
     @Bean("search_EnqueueingSessionProcessingTrigger")
+    @ConditionalOnProperty(name = "jmix.search.use-default-enqueueing-session-processing-quartz-configuration", matchIfMissing = true)
     Trigger enqueueingSessionProcessingTrigger(@Qualifier("search_EnqueueingSessionProcessingJob") JobDetail enqueueingSessionProcessingJob) {
         String cron = searchProperties.getEnqueueingSessionProcessingCron();
         log.info("Schedule Enqueueing Session processing using default configuration with CRON expression '{}'", cron);
         return TriggerBuilder.newTrigger()
-                .withIdentity("EnqueueingSessionProcessingTrigger")
+                .withIdentity(JOB_TRIGGER_NAME)
                 .forJob(enqueueingSessionProcessingJob)
                 .startNow()
                 .withSchedule(CronScheduleBuilder.cronSchedule(cron))
                 .build();
     }
+
+    @PostConstruct
+    void cleanJob() {
+        if (!searchProperties.isUseDefaultEnqueueingSessionProcessingQuartzConfiguration()) {
+            JobKey jobKey = JobKey.jobKey(JOB_NAME, JOB_GROUP);
+            Scheduler scheduler = applicationContext.getBean(Scheduler.class);
+            try {
+                scheduler.deleteJob(jobKey);
```

Comment:
Check if job exists via `scheduler.checkExists(jobKey)` to prevent unnecessary delete query execution.

Reply:
done